File: frontend/src/app/models/Category.ts
import { SubCategory } from "./SubCategory";

export class Category{

    name?: String
    id?: number;
    subcategories?: SubCategory[];
    editable?: boolean;

    constructor({name,id,subcategories}:{name?: String, id?: number, subcategories?: SubCategory[]}) {
        this.name=name;
        this.id=id;
        this.subcategories=subcategories;
        this.editable=false;
    }

}

File: backend/CoreCuestionariosOIJ/src/CuestionariosEntidades/Models/Category.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace CuestionariosEntidades.Models
{
    public class Category
    {
        public int? Id { get; set; }
        public string? Name { get; set; } = string.Empty;

        // EF Relation
        public IEnumerable<SubCategory>? SubCategories { get; set; }
        // EF Relation
        public IEnumerable<Question>? Questions { get; set; }

    }
}
